python/pyarrow/tests/test_compute.py

```diff
@@ -85,6 +85,8 @@ def test_exported_functions():
     functions = exported_functions
     assert len(functions) >= 10
     for func in functions:
+        if func in (pc.index_in_meta_binary, pc.is_in_meta_binary):
+            continue
```

Comment:
Is there a particular reason for this?

Reply:
Have to still look into this, but so for those two functions, the test was failing. 

So the reason it was failing is because the metadata of the function is not fully consistent:
```
In [13]: func = pc.function_registry().get_function("index_in_meta_binary")

In [14]: pc._get_arg_names(func)
Out[14]: []

In [15]: func.arity
Out[15]: 2
```

I was using the arg_names for checking arity, but so in the case of the meta functions that apparently is not the same. I suppose that is because the `index_in_meta_binary` has no `doc` defined.

Reply:
Hmm, if the doc is wrong or missing, that should ideally be fixed on the C++ side.

Reply:
Cleaned up in the last commit by using `func.arity` instead of `len(arg_names)`

Reply:
Indeed, I solved it differently (which is cleaner anyway), but can still use a docstring on the C++ side

Reply:
Can you open a JIRA for the missing/inaccurate C++ docs?

Reply:
Done: https://issues.apache.org/jira/browse/ARROW-14751